src/serviceDialog.js

```diff
@@ -12,8 +12,67 @@
     ResponseType,
     EntryIconPosition,
   } = imports.gi.Gtk;
+
+  const GLib = imports.gi.GLib;
+  const GObj = imports.gi.GObject;
+  const Gtk = imports.gi.Gtk;
+
   const { uuid_string_random } = imports.gi.GLib;
 
+  const openIconChooserDialog = function() {
+    const filter = new Gtk.FileFilter();
+    filter.add_mime_type("image/png");
+
+    const chooser = new Gtk.FileChooserDialog({
+      action: Gtk.FileChooserAction.OPEN,
+      filter: filter,
+      select_multiple: false,
+      //transient_for: this.window,
+      title: "Open",
+    });
+
+    // Without setting a current folder, folders won't show its contents
+    chooser.set_current_folder(GLib.get_home_dir());
+
+    chooser.add_button("Cancel", Gtk.ResponseType.CANCEL);
+    chooser.add_button("OK", Gtk.ResponseType.OK);
+
+    // This is to add the 'combo' filtering options
+    const store = new Gtk.ListStore();
+    store.set_column_types([GObj.TYPE_STRING, GObj.TYPE_STRING]);
+    store.set(store.append(), [0, 1], ["png", "image/png"]);
+    store.set(store.append(), [0, 1], ["jpg", "image/jpeg"]);
+    store.set(store.append(), [0, 1], ["svg", "image/svg+xml"]);
+
+    const combo = new Gtk.ComboBox({ model: store });
+    const renderer = new Gtk.CellRendererText();
+    combo.pack_start(renderer, false);
+    combo.add_attribute(renderer, "text", 1);
+    combo.set_active(0);
+    combo.connect("changed", widget => {
+      const model = widget.get_model();
+      const active = widget.get_active_iter()[1];
+
+      const text = model.get_value(active, 1);
+
+      const filter = new Gtk.FileFilter();
+      filter.add_mime_type(text);
+
+      chooser.set_filter(filter);
+    });
+    chooser.set_extra_widget(combo);
```

Comment:
The combo is not needed, it makes things more complicated for both the code and the user.